services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -158,39 +195,119 @@ class HankeServiceImpl (@Autowired val hankeRepository: HankeRepository) : Hanke
-            hanke.tyomaaKoko?.let { entity.tyomaaKoko = hanke.tyomaaKoko }
-
-            // Assuming the incoming date, while being zoned date and time, is in UTC and time value can be simply dropped here.
-            // Note, .toLocalDate() does not do any time zone conversion.
-            hanke.haittaAlkuPvm?.let { entity.haittaAlkuPvm = hanke.haittaAlkuPvm?.toLocalDate() }
-            hanke.haittaLoppuPvm?.let { entity.haittaLoppuPvm = hanke.haittaLoppuPvm?.toLocalDate() }
-            hanke.kaistaHaitta?.let { entity.kaistaHaitta = hanke.kaistaHaitta }
-            hanke.kaistaPituusHaitta?.let { entity.kaistaPituusHaitta = hanke.kaistaPituusHaitta }
-            hanke.meluHaitta?.let { entity.meluHaitta = hanke.meluHaitta }
-            hanke.polyHaitta?.let { entity.polyHaitta = hanke.polyHaitta }
-            hanke.tarinaHaitta?.let { entity.tarinaHaitta = hanke.tarinaHaitta }
+        internal fun createSeparateYhteystietolistsFromEntityData(hanke: Hanke, hankeEntity: HankeEntity) {
+
+            hankeEntity.listOfHankeYhteystieto?.forEach { hankeYhteysEntity ->
+                var hankeYhteystieto = createHankeYhteystietoDomainObjectFromEntity(hankeYhteysEntity)
+
+                if (hankeYhteysEntity.contactType.equals(ContactType.OMISTAJA))
+                    hanke.omistajat.add(hankeYhteystieto)
+                if (hankeYhteysEntity.contactType.equals(ContactType.TOTEUTTAJA))
+                    hanke.toteuttajat.add(hankeYhteystieto)
+                if (hankeYhteysEntity.contactType.equals(ContactType.ARVIOIJA))
+                    hanke.arvioijat.add(hankeYhteystieto)
+            }
         }
+
+        internal fun createHankeYhteystietoDomainObjectFromEntity(hankeYhteystietoEntity: HankeYhteystietoEntity): HankeYhteystieto {
+            var createdAt: ZonedDateTime? = null
+
+            if (hankeYhteystietoEntity.createdAt != null)
+                createdAt = ZonedDateTime.of(hankeYhteystietoEntity.modifiedAt, TZ_UTC)
+
+            var modifiedAt: ZonedDateTime? = null
+            if (hankeYhteystietoEntity.modifiedAt != null)
+                modifiedAt = ZonedDateTime.of(hankeYhteystietoEntity.modifiedAt, TZ_UTC)
+
+            return HankeYhteystieto(
+                    id = hankeYhteystietoEntity.id,
+                    etunimi = hankeYhteystietoEntity.etunimi,
+                    sukunimi = hankeYhteystietoEntity.sukunimi,
+                    email = hankeYhteystietoEntity.email,
+                    puhelinnumero = hankeYhteystietoEntity.puhelinnumero,
+                    organisaatioId = hankeYhteystietoEntity.organisaatioid,
+                    organisaatioNimi = hankeYhteystietoEntity.organisaationimi,
+                    osasto = hankeYhteystietoEntity.osasto,
+                    createdBy = hankeYhteystietoEntity.createdByUserId?.toString() ?: "",
+                    modifiedBy = hankeYhteystietoEntity.modifiedByUserId?.toString() ?: "",
+                    createdAt = createdAt,
```

Comment:
Shorter, simpler:
if (hankeYhteystietoEntity.createdAt != null) ZonedDateTime.of(hankeYhteystietoEntity.modifiedAt, TZ_UTC) else null,

and same for modifiedAt..

Reply:
Well this would be the one reason not to have all the fields in the constructor.. but since they are and this is in fact contructor initialization, the example you have there won't work as it is.